### PR TITLE
chore: use starknetkit for wallet connection

### PR DESCRIPTION
**Pull request by Darlington02**
Update frontend to use starknetkit for a more robust connect modal.

### CODE REVIEW

File: frontend/package-lock.json
Comment (0xChqrles) [inline frontend/package-lock.json:1]: We're using yarn for this project, pls just remove this lockfile and it's all good !

**Review by 0xChqrles — CHANGES_REQUESTED** [review]
Thanks 🙏

File: frontend/package-lock.json
Comment (Darlington02) [reply frontend/package-lock.json:1]: Ha! Sorry about that. Updating in a min

Comment (Darlington02) [reply frontend/package-lock.json:1]: Done:)

**Review by 0xChqrles — APPROVED** [review]
lgtm

**Pull request merged**